mrp_client.py

```diff
@@ -69,6 +74,39 @@ def preseed(self, command, preseed_name, preseed_file, preseed_desc,
             self.log.fatal(err)
             exit(1)
 
+    def machine_control(self, machine_name, action, preseed_name, initrd_desc, kernel_desc,
+                             kernel_opts, arch, subarch):
+        try:
+            state = StateControl(self.urlhandler, machine_name)
+
+            if action == 'getparams':
+                machine_state = state.get_state()
+                print(self.__print_machine_state(machine_state))
+            else:
+                rc = state.set_state(arch, subarch, initrd_desc, kernel_desc,
+                                     kernel_opts, preseed_name)
+                self.log.debug(rc)
+            if action == 'provision':
+                rc = state.provision()
+                self.log.debug(rc)
+
+        except Exception as err:
+            self.log.fatal(err)
+            exit(1)
+
+    def __print_machine_state(self, machine_state):
```

Comment:
why not just print directly from here?
a simple for-each would do:
```
for key, value in machine_state:
    print("%: %\n" % key, str(value))
```

Also, why two underscores?

Reply:
True that's much simpler silly me !

In Python it means private !

Reply:
In Python, one underscore is private, two has special mangling. We don't need two, just one.